tree/tree/src/TBasket.cxx

```diff
@@ -725,20 +740,48 @@ void TBasket::Reset()
          }
       }
    }
-   /*
-      Philippe has asked us to keep this turned off until we finish memory fragmentation studies.
-   // If fBufferRef grew since we last saw it, shrink it to 105% of the occupied size
-   if (curSize > fLastWriteBufferSize) {
-      if (newSize == -1) {
-         newSize = Int_t(1.05*Float_t(fBufferRef->Length()));
+   // If fBufferRef grew since we last saw it, shrink it to "target memory ratio" of the occupied size
+   // This discourages us from having poorly-occupied buffers on branches with little variability.
+   //
+   // Does not help protect against a burst in event sizes, but does help in the cases where the basket
+   // size jumps from 4MB to 8MB while filling the basket, but we only end up utilizing 4.1MB.
+   //
+   // The above code block is meant to protect against extremely large events.
+
+   Float_t target_mem_ratio = fBranch->GetTree()->GetTargetMemoryRatio();
+   Int_t max_size = std::max(fLastWriteBufferSize[0], std::max(fLastWriteBufferSize[1], fLastWriteBufferSize[2]));
```

Comment:
for consistency with the res of the code use TMath::Max ...

Reply:
I lost track.  Why are we trying to get TMath dependencies out of TTree?

Reply:
Maybe I confused this with another aspect of Oksana's work - but the idea was to simplify some of the interdependencies between modules as part of the "ROOT base" effort.

Anyhow, I can make the change here.

Reply:
TMath::Max is in TMathCore.h (so in core) ...  For consistency's sake can you replace all the std::max with TMath::max (or I guess all the TMath::max with std::max ...)